src/main/java/th/ac/kmitl/science/comsci/example/nameprinters/NamePrinterLoader.java

```diff
@@ -0,0 +1,56 @@
+package th.ac.kmitl.science.comsci.example.nameprinters;
+
+import org.reflections.Reflections;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+public class NamePrinterLoader implements Iterable<NamePrinter> {
+
+    //Singleton implementation
+    private static NamePrinterLoader instance;
+    private final String packageIdentifier;
+    private List<NamePrinter> namePrinters;
+    private NamePrinterLoader() {
+        packageIdentifier = getClass().getPackage().getName();
+
+        try {
+            namePrinters = getAllNamePrinters();
+        } catch (Exception e) {
+            e.printStackTrace();
+            namePrinters = null;
+        }
+    }
+
+    public static NamePrinterLoader getInstance() {
+        if (instance == null)
+            instance = new NamePrinterLoader();
+
```

Comment:
Why we use singleton here ?
Don't be scared, just asking for reason.

Reply:
@mahasak because all `NamePrinter`s have to be in `th.ac.kmitl.science.comsci.example.nameprinters` package and it's not possibly append the other `NamePrinter` to this package at runtime so loading all `NamePrinter` at once would be nicer.